src/Sentry/Protocol/Metrics/DistributionMetric.cs

```diff
@@ -0,0 +1,35 @@
+using Sentry.Extensibility;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Protocol.Metrics;
+
+/// <summary>
+/// Distributions track a list of values over time in on which you can perform aggregations like max, min, avg.
+/// </summary>
+internal class DistributionMetric : Metric
+{
+    public DistributionMetric()
+    {
+        Value = new List<double>();
+    }
+
+    public DistributionMetric(string key, double value, MeasurementUnit? unit = null,
+        IDictionary<string, string>? tags = null, DateTime? timestamp = null)
+        : base(key, unit, tags, timestamp)
+    {
+        Value = new List<double>() { value };
+    }
+
+    public IList<double> Value { get; set; }
+
+    public override void Add(double value)
+    {
+        Value.Add(value);
+    }
+
+    protected override void WriteValues(Utf8JsonWriter writer, IDiagnosticLogger? logger) =>
+        writer.WriteArrayIfNotEmpty<double>("value", Value, logger);
+
+    protected override IEnumerable<IConvertible> SerializedStatsdValues()
+        => Value.Select(v => (IConvertible)v);
```

Comment:
if we'll have a lot of those, a `foreach` here instead avoids an allocation I believe (but we can benchmark this and improve in a follow up PR)

Reply:
I'll change it to Cast<T> actually... that's all the select clause is doing anyway.